fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobManager.java

```diff
@@ -0,0 +1,278 @@
+        try {
+            boolean registered = registerOverwriteJob(job);
+            if (!registered) {
+                LOG.warn("register insert overwrite job:{} failed", job.getJobId());
+                throw new RuntimeException("register insert overwrite job failed");
+            }
+            job.run();
+        } finally {
+            deregisterOverwriteJob(job.getJobId());
+        }
+    }
+
+    public void registerOverwriteJobTxn(long jobId, long txnId) {
+        lock.writeLock().lock();
+        try {
+            jobToTxnId.put(jobId, txnId);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public boolean registerOverwriteJob(InsertOverwriteJob job) {
+        lock.writeLock().lock();
+        try {
+            if (overwriteJobMap.containsKey(job.getJobId())) {
+                LOG.warn("insert overwrite job:{} is running", job.getJobId());
+                return false;
+            }
+            // check whether there is a overwrite job running in partitions
+            if (hasRunningOverwriteJob(-1, job.getTargetTableId(), job.getOriginalTargetPartitionIds())) {
+                LOG.warn("table:{} has running overwrite jobs", job.getTargetTableId());
+                return false;
+            }
+            overwriteJobMap.put(job.getJobId(), job);
+            List<Long> runningPartitions = partitionsWithOverwrite.getOrDefault(job.getTargetTableId(), Lists.newArrayList());
+            if (job.getOriginalTargetPartitionIds() != null) {
+                runningPartitions.addAll(job.getOriginalTargetPartitionIds());
+            }
+            partitionsWithOverwrite.put(job.getTargetTableId(), runningPartitions);
+            jobNum++;
+            return true;
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public boolean deregisterOverwriteJob(long jobid) {
+        lock.writeLock().lock();
+        try {
+            if (!overwriteJobMap.containsKey(jobid)) {
+                return true;
+            }
+            InsertOverwriteJob job = overwriteJobMap.get(jobid);
+            jobToTxnId.remove(jobid);
+            List<Long> partitionIds = partitionsWithOverwrite.get(job.getTargetTableId());
+            if (partitionIds != null) {
+                partitionIds.removeAll(job.getOriginalTargetPartitionIds());
+                if (partitionIds.isEmpty()) {
+                    partitionsWithOverwrite.remove(job.getTargetTableId());
+                }
+            } else {
+                partitionsWithOverwrite.remove(job.getTargetTableId());
+            }
+            overwriteJobMap.remove(jobid);
+            jobNum--;
+            return true;
+        } catch (Exception e) {
+            LOG.warn("deregister overwrite job failed", e);
+            throw e;
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    public boolean hasRunningOverwriteJob(long txnId, long tableId, List<Long> partitions) {
```

Comment:
Will we support insert into many partitions of one table at the same time?

Reply:
yes, we do support this function.